File: 15.oops/7.single-inheritance.js
class Animal {
  #name;
  #species;

  constructor(name, species) {
    this.#name = name;
    this.#species = species;
  }

  getName() {
    return this.#name;
  }

  getSpecies() {
    return this.#species;
  }
}

class Dog extends Animal {
  #breed;
  constructor(species, breed, name) {
    super(name, species);
    this.#breed = breed;
  }

  getBreed() {
    return this.#breed;
  }
}

const d1 = new Dog("Dog", "German shepherd", "Rocky");

console.log(d1.getSpecies());
console.log(d1.getBreed());
console.log(d1.getName());

File: 15.oops/7single-inheritance.cpp
#include <iostream>

using namespace std;

class Animal {
  string name;
  string species;

 public:
  Animal(string name, string species) {
    this->name = name;
    this->species = species;
  }

  string get_name() { return this->name; }

  string get_species() { return this->species; }
};

class Dog : public Animal {
  string breed;

 public:
  Dog(string breed, string name, string species) : Animal(name, species) {
    this->breed = breed;
  }

  string get_breed() { return this->breed; }
};

int main() {
  Dog d1("German shepheard", "Rocky", "Dog");

  cout << d1.get_species() << endl;
  cout << d1.get_breed() << endl;
  cout << d1.get_name() << endl;

  return 0;
}